README.md

```diff
@@ -0,0 +1,146 @@
+SELinux Generic Monitor Interface to create simple SELinux modules for monitoring applications
+==========================================================
+https://github.com/Laurent-Gaillard/selinux_generic_monitor
+
+## Introduction
+
+This SELinux interface was created to help people develop simple SELinux modules for
+monitoring applications.
+
+The file exposes multiple interface named after their purpose 
+
+When used correctly, this SELinux interface will avoid you a lot of work by calling 
+simple interfaces.
+
+## Prerequisites
+
+You will absolutely need to install selinux-policy-devel package to compile SELinux modules.
+
+## How to use this SELinux module
+
+This interface file must be placed into a subdirectory of */usr/share/selinux/devel/include/*
+You can either:  
+ - put it in the apps subdirectory  
+ - create a subdirectory of your own
+
+Once the SELinux interface is in place, you will be able to use the interfaces in your own 
+modules and compile them.
+
+## Templates / Interfaces explanation
+
+"*Domain subname*" stands for the name of the monitoring application / purpose, ...
+
+### generic_mon_app_domain
+- *type*: **template**
+- *description*:  ****This template creates a domain, types, and rules to be used by a monitoring application.**
+- *parameters*:
+    - **domain subname**
+     
+
+### generic_mon_listening_socket:
+- *type*: **interface**
+- *description*:  **Authorize to create a listening socket.**
+- *parameters*:
+    - **domain subname**
+
+### generic_mon_connecting_socket:
+- *type*: **interface**
+- *description*:  **Authorize to connect to socket.**
+- *parameters*: The domain subname to be authorized to connect to socket.
+    - **domain subname**
+
+### generic_mon_commom_access
+- *type*: **interface**
+- *description*:  **Authorize to get some basic infos for monitoring tools (processes, fs info).**
+- *parameters*: 
+    - **domain subname**
+
+### generic_mon_add_netadmin_capability
+- *type*: **interface**
+- *description*:  **Authorize to manage all networking configurations and modifications on self. 
+                   Use carefully
+- *parameters*: 
+    - **domain subname**
+
+### generic_mon_add_execmem_capability
```

Comment:
Why not create a boolean in the main template for that ?

Reply:
Because I didn't find creation and use of boolean inside other if files. So I thought this wasn't a good idea.
But I think it can be done.
You mean for both booleans?

Reply:
If this is ok for you, can you please validate the PR?

Reply:
I finally created the two booleans. Need validation please.